esp-hal-common/src/uart.rs

```diff
@@ -483,6 +489,12 @@ where
         serial.change_parity(config.parity);
         serial.change_stop_bits(config.stop_bits);
         serial.change_baud(config.baudrate, clocks);
+        // Should this be done ? Or configuration option? Setting it stops uart from
+        // storing data when data is wrong
+        T::register_block()
+            .conf0()
+            .modify(|_, w| unsafe { w.err_wr_mask().set_bit() });
+        // TODO reset? https://github.com/espressif/esp-idf/blob/4d90eedb6ef32ff949c6f298a3d845f9f07bac92/components/hal/esp32s3/include/hal/uart_ll.h#L112
```

Comment:
I think this reset is indeed required, could you implement it here?

Reply:
I think it would need some more work. Would probably create a new ticket/issue for it if that is ok?

Reply:
Sure! Sounds good to me :)